src/ros_filter.cpp

```diff
@@ -800,21 +799,23 @@ namespace RobotLocalization
         ROS_INFO_STREAM("use_control is set to true, but no deceleration_limits specified. Will use acceleration "
           "limits.");
         decelerationLimits = accelerationLimits;
-        useAccelLimits = true;
       }
 
-      decelerationGains.resize(TWIST_SIZE, 1.0);
       if(nhLocal_.getParam("deceleration_gains", decelerationGains))
       {
-        if(decelerationGains.size() != TWIST_SIZE)
+        const size_t size = decelerationGains.size();
+        if(size != TWIST_SIZE)
         {
-          ROS_ERROR_STREAM("Acceleration gain configuration must be of size " << TWIST_SIZE <<
-            ". Provided config was of size " << decelerationLimits.size() << ". All gains will be assumed to be 1.");
+          ROS_ERROR_STREAM("Deceleration gain configuration must be of size " << TWIST_SIZE <<
+            ". Provided config was of size " << size << ". All gains will be assumed to be 1.");
+          std::fill_n(decelerationGains.begin(), size, 1.0);
+          decelerationGains.resize(TWIST_SIZE, 1.0);
```

Comment:
Would it make sense to reverse lines 811 and 812? That way, we avoid filling a very large vector if `decelerationGains.size()` is massive (I can't see this being the case, but it can't hurt to be safe). Also, your logic here is closer to what I would have expected for your changes on line 783, but you aren't doing the `resize` there. I may be missing your intent, though.